src/services/driver.py

```diff
@@ -0,0 +1,40 @@
+from datetime import datetime, date, timedelta
+
+from src.model import Driver, Schedule
+
+class DriverService(object):
+    async def get_with_schedules_by_week(self, driver_id: int, day_of_the_week: date):
+        first_day_of_the_week = day_of_the_week - timedelta(days=day_of_the_week.weekday() + 1)
+        last_day_of_the_week = first_day_of_the_week + timedelta(days=7)
+
+        driver = await Driver.objects.get_or_none(id=driver_id)
+        driver.schedules = await Schedule.objects.filter(driver=driver, begin__gte=first_day_of_the_week, end__lt=last_day_of_the_week).select_related(Schedule.bus).all()
+
+        return driver
+
+    async def get_top_drivers(self, weekDateBegin: date, weekDateEnd: date):
+        first_day = weekDateBegin - timedelta(days=weekDateBegin.weekday() + 1)
+        last_day = weekDateEnd - timedelta(days=weekDateEnd.weekday() + 1) + timedelta(days=7)
+
+        drivers = await Driver.objects.database.fetch_all("""
```

Comment:
@ricardovsilva ,What was the reasoning behind using SQL vs the form in this situation?

Reply:
Because the documentation of `ormar` was not so clear about how to filter the query based on the field of another table. So I preferred to use plain SQL. Also, IMHO, the plain SQL for this case is clearer and more straightforward than the ORM style.